File: CS/NotifyIconService/ViewModels/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.POCO;
using DevExpress.Xpf.Core;

namespace NotifyIconService.ViewModels {
    [POCOViewModel]
    public class MainViewModel {

        public MainViewModel() {
            IsConnected = null;
        }

        protected ICurrentWindowService CurrentWindowService { get { return this.GetService<ICurrentWindowService>(); } }
        protected INotifyIconService NotifyIconService { get { return this.GetService<INotifyIconService>(); } }
        protected IMessageBoxService MessageBoxService { get { return this.GetService<IMessageBoxService>(); } }

        public void ActivateMainWindow() {
            CurrentWindowService.SetWindowState(System.Windows.WindowState.Normal);
            CurrentWindowService.Activate();
        }
        public void Connect() {
            NotifyIconService.SetStatusIcon("connected");
            IsConnected = true;
        }
        public bool CanConnect() {
            return IsConnected != true;
        }

        public void Disconnect() {
            NotifyIconService.SetStatusIcon("disconnected");
            IsConnected = false;
        }
        public bool CanDisconnect() {
            return IsConnected == true;
        }

        public void Reset() {
            NotifyIconService.ResetStatusIcon();
            IsConnected = null;
        }

        public void About() {
            MessageBoxService.ShowMessage("This example demonstrates how to use NotifyIconService ", "About");
        }

        public virtual bool? IsConnected { get; set; }
    }
}

File: VB/NotifyIconService/ViewModels/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Xpf.Core

Namespace NotifyIconService.ViewModels

    <POCOViewModel>
    Public Class MainViewModel

        Public Sub New()
            IsConnected = Nothing
        End Sub

        Protected ReadOnly Property CurrentWindowService As ICurrentWindowService
            Get
                Return Me.GetService(Of ICurrentWindowService)()
            End Get
        End Property

        Protected ReadOnly Property NotifyIconService As INotifyIconService
            Get
                Return Me.GetService(Of INotifyIconService)()
            End Get
        End Property

        Protected ReadOnly Property MessageBoxService As IMessageBoxService
            Get
                Return Me.GetService(Of IMessageBoxService)()
            End Get
        End Property

        Public Sub ActivateMainWindow()
            CurrentWindowService.SetWindowState(System.Windows.WindowState.Normal)
            CurrentWindowService.Activate()
        End Sub

        Public Sub Connect()
            NotifyIconService.SetStatusIcon("connected")
            IsConnected = True
        End Sub

        Public Function CanConnect() As Boolean
            Return IsConnected <> True
        End Function

        Public Sub Disconnect()
            NotifyIconService.SetStatusIcon("disconnected")
            IsConnected = False
        End Sub

        Public Function CanDisconnect() As Boolean
            Return IsConnected = True
        End Function

        Public Sub Reset()
            NotifyIconService.ResetStatusIcon()
            IsConnected = Nothing
        End Sub

        Public Sub About()
            MessageBoxService.ShowMessage("This example demonstrates how to use NotifyIconService ", "About")
        End Sub

        Public Overridable Property IsConnected As Boolean?
    End Class
End Namespace
